### PR TITLE
Fix linter

### PR DESCRIPTION
**Pull request by ocefpaf**
~~Merge only after #126.~~

This one is ready for review!

### CODE REVIEW

File: gsw/geostrophy.py [hunk 1]
```diff
@@ -295,7 +295,7 @@ def geostrophic_velocity(geo_strf, lon, lat, p=0, axis=0):
 
     if lon.shape != lat.shape or lon.ndim != 1:
         raise ValueError('lon, lat must be 1-D and matching; found shapes'
-                         ' %s and %s' % (lon.shape, lat.shape))
+                         ' {} and {}'.format(lon.shape, lat.shape))
```
Comment (efiring) [inline gsw/geostrophy.py:298]: Might as well go all the way to an f-string.

File: gsw/geostrophy.py [hunk 2]
```diff
@@ -295,7 +295,7 @@ def geostrophic_velocity(geo_strf, lon, lat, p=0, axis=0):
...
 
     if geo_strf.ndim not in (1, 2):
         raise ValueError('geo_strf must be 1-D or 2-d; found shape %s'
```
Comment (efiring) [inline gsw/geostrophy.py:301]: I wonder why the old percent syntax is changed above, but not here?
There's actually nothing wrong with the old  syntax; it isn't going to go
away.  But in most cases, f-strings are more readable.  I would certainly
use them for new code.

File: pyproject.toml [hunk 3]
```diff
@@ -94,6 +93,14 @@ exclude = [
...
+"docs/conf.py" = [
   "A001",
 ]
+"gsw/_fixed_wrapped_ufuncs.py" = [
+  "F403",
+  "F405",
+]
+
+"gsw/_utilities.py" = [
+  "B904",
```
Comment (efiring) [inline pyproject.toml:105]: For each of these skips, please add a short comment identifying what the rule addresses.

**Review by efiring — CHANGES_REQUESTED** [review]
Minor questions; overall, I like the cleanups.

File: gsw/geostrophy.py (hunk 2 above)
Comment (efiring) [reply gsw/geostrophy.py:301]: This comment somehow got connected to the wrong line.  A few lines below here there is an unchanged line that has the percent syntax, and presumably ruff missed it.

File: gsw/geostrophy.py (hunk 1 above)
Comment (ocefpaf) [reply gsw/geostrophy.py:298]: Will do. I'm not sure why pyupgrade, even though it is set to py38, stopped at `.format`.

File: gsw/geostrophy.py (hunk 2 above)
Comment (ocefpaf) [reply gsw/geostrophy.py:301]: Probably a bug in pyupgrade via ruff. I'll investigate and report upstream.

Comment (ocefpaf) [reply gsw/geostrophy.py]: Looks like we are in the situation in the note here: https://github.com/asottile/pyupgrade#f-strings

File: pyproject.toml (hunk 3 above)
Comment (ocefpaf) [reply pyproject.toml]: Done!

**Review by efiring — APPROVED** [review]

File: requirements-dev.txt [hunk 4]
```diff
@@ -6,6 +6,5 @@ setuptools_scm
 sphinx
 sphinx_rtd_theme
 twine
-xarray
```
[the remark this answers is not in the document]
Comment (DocOtak) [reply requirements-dev.txt:9]: We test the full (apparent) API surface of GSW-Python in in gsw_xarray, so I wouldn't be too worried about breakages.

**Pull request merged**